### ISSUE TITLE
Allow ShouldSerialize pattern for Nullables

### ISSUE THREAD

**Issue by gambrose** [mganss/XmlSchemaClassGenerator#144]
An alternative pattern for dealing with nullables in XML serialisation is to add a method named `ShouldSerialize` then the property name.  This stops `XmlSerializer` from outputting nulls.

I prefer this approach as one extra method is less "polluting" than two extra properties.

``` c#
[System.Xml.Serialization.XmlAttributeAttribute("id", Form = System.Xml.Schema.XmlSchemaForm.Unqualified)]
public System.Nullable<int> Id { get; set; }

public bool ShouldSerializeId() => Id.HasValue;
```

**Comment by gambrose** [comment]
There does not appear to be any tests covering the existing nullable pattern. Could you offer some advice as how best to create these tests?
I would like to ensure I don't break existing functionality before I add the new. I was going to add an config option to allow which strategy to use for handling nullables.

**Comment by mganss** [comment]
There are no explicit tests but there are implicit tests through `XmlSampleGenerator`:
https://github.com/mganss/XmlSchemaClassGenerator/blob/7dd924be1c1bfe8c8ff02b6cfd591040f94f97f0/XmlSchemaClassGenerator.Tests/XmlTests.cs#L240

You could add an explicit test using a small schema that has an optional attribute and check that an object round-trips to/from XML correctly for both cases.

**Comment by mganss** [comment]
Fixed by #145
